src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3299,6 +3301,277 @@ tx_gateway_frontend::get_all_transactions() {
+      || tm_transaction::tx_status::preparing) {
+        // We can just ignore this statuses
+        co_return tx;
+    } else if (tx.status == tm_transaction::tx_status::ready) {
+        auto r1 = co_await fetch_tx(tx.id, tx.etag);
+        if (!r1) {
+            vlog(
+              txlog.warn,
+              "can't find old tx:{} etag:{} during describe_tx call",
+              tx.id,
+              tx.etag);
+            // Return possibly stale data
+            co_return tx;
+        }
+
+        auto old_tx = r1.value();
+
+        if (old_tx.status == tm_transaction::tx_status::ongoing) {
+            if (old_tx.pid != tx.pid) {
+                vlog(
+                  txlog.warn,
+                  "fetched pid:{} doesn't match local tx's pid:{}",
+                  old_tx.pid,
+                  tx.pid);
+                co_return tx_errc::unknown_server_error;
+            }
+        } else if (old_tx.status == tm_transaction::tx_status::ready) {
+            if (old_tx.tx_seq > tx.tx_seq) {
+                vlog(
+                  txlog.warn,
+                  "Fethced tx_seq:{} is bigger local tx's tx_seq:{}",
+                  old_tx.tx_seq,
+                  tx.tx_seq);
+                co_return tx_errc::unknown_server_error;
+            }
+            old_tx = tx;
+        } else if (old_tx.tx_seq < tx.tx_seq) {
+            old_tx = tx;
+        } else {
+            vlog(
+              txlog.warn,
+              "Unexpected fetched tx. status:{} tx_seq:{}",
+              old_tx.get_status(),
+              tx.tx_seq);
+            co_return tx_errc::unknown_server_error;
+        }
+
+        old_tx.etag = term;
+        auto r2 = co_await stm->update_tx(old_tx, term);
+        if (!r2) {
+            vlog(txlog.info, "got {} on updating fetched tx", r2.error());
+            co_return tx_errc::not_coordinator;
+        }
+        tx = r2.value();
+    } else if (tx.status == tm_transaction::tx_status::killed) {
+        // We can just return current tx, becasue old tx also should be expired
+        co_return tx;
+    } else if (
+      tx.status == tm_transaction::tx_status::prepared
+      || tx.status == tm_transaction::tx_status::aborting) {
+        auto r1 = co_await fetch_tx(tx.id, tx.etag);
+        if (!r1) {
+            vlog(
+              txlog.warn,
+              "can't find old tx:{} etag:{} during describe_tx call",
+              tx.id,
+              tx.etag);
+            // Return possibly stale data
+            co_return tx;
+        }
+
+        auto old_tx = r1.value();
+
+        if (old_tx.pid != tx.pid) {
+            vlog(
+              txlog.warn,
+              "fetched pid:{} doesn't match local tx's pid:{}",
+              old_tx.pid,
+              tx.pid);
+            co_return tx_errc::unknown_server_error;
+        }
+
+        if (old_tx.status == tm_transaction::tx_status::ongoing) {
+            if (old_tx.tx_seq() == tx.tx_seq()) {
+                co_return tx;
```

Comment:
we still need to update etag, let's do `old_tx = tx` and update it below just like you did with ready